rotkehlchen/chain/ethereum/modules/liquity/trove.py

```diff
@@ -287,6 +294,65 @@ def get_positions(
                     )
         return data
 
+    def get_stability_pool_positions(
+        self,
+        addresses: List[ChecksumEvmAddress],
+    ) -> Dict[ChecksumEvmAddress, Dict[str, FVal]]:
+        """Get current deposit in the stability pool and current ETH/LQTY gains"""
+        # make a copy of the list to avoid modifications in the list that is passed as argument
+        addresses = addresses.copy()
+        proxied_addresses = self._get_accounts_having_proxy()
+        addresses += proxied_addresses.values()
+
+        # Build the calls that need to be made in order to get the status in the SP
+        calls = []
+        for address in addresses:
+            calls.append(
+                (LIQUITY_STABILITY_POOL.address, self.stability_pool_contract.encode(method_name='getDepositorETHGain', arguments=[address])),  # noqa: E501
+            )
+            calls.append(
+                (LIQUITY_STABILITY_POOL.address, self.stability_pool_contract.encode(method_name='getDepositorLQTYGain', arguments=[address])),  # noqa: E501
+            )
+            calls.append(
+                (LIQUITY_STABILITY_POOL.address, self.stability_pool_contract.encode(method_name='getCompoundedLUSDDeposit', arguments=[address])),  # noqa: E501
+            )
+
+        try:
+            outputs = self.ethereum.multicall_2(
+                require_success=False,
+                calls=calls,
+            )
+        except (RemoteError, BlockchainQueryError) as e:
+            log.error(f'Failed to query stability pool information due to: {str(e)}')
+            self.msg_aggregator.add_error(
```

Comment:
is the msg aggregator error needed? Or better question. Even this line `log.error(f'Failed to query stability pool information due to: {str(e)}')` is not big and contains no sensitive info. So coudn't it all just be a msg_aggregator.add_error? (since there is a log error in there already?). There is nothing more to see later in the logs.